app/domain/authentication/authn_oidc/authenticator.rb

```diff
@@ -0,0 +1,84 @@
+module Authentication
+  module AuthnOidc
+    class AuthenticationError < RuntimeError; end
+    class NotFoundError < RuntimeError; end
+    class OIDCConfigurationError < RuntimeError; end
+    class OIDCAuthenticationError < RuntimeError; end
+
+    class Authenticator
+
+      def initialize(env:)
+        @env = env
+      end
+
+      def valid?(input)
+        @authenticator_name = input.authenticator_name
+        @service_id = input.service_id
+        @conjur_account = input.account
+        @request_body = input.request.body.read
+
+        verify_service_enabled
+
+        oidc_authn_service = AuthenticationService.new(service.identifier, conjur_account)
```

Comment:
`AuthenticationService` should be injected.  See my other command where I discuss `CommandClass`

Reply:
I added a ticket for this, we'll get to it after the split